### ISSUE TITLE
Have an option to preserve world scale, position and rotation when reparenting

### ISSUE THREAD

**Issue by yaustar** [playcanvas/editor#150]
Forum thread here: https://forum.playcanvas.com/t/entities-resize-when-re-parenting-in-editor-how-do-i-get-them-to-stop/14838/2

It would be useful for world scale in particular to be preserved when reparenting.

**Comment by Maksims** [comment]
I don't think it is actually possible. Scale can affect model by axes, and ancestors with different rotations, will lead to all sorts of skews and rotations.
This cannot be replicated by own "scale", as there are so many different options, not like with position and rotation.
There is also no "world scale".

**Comment by yaustar** [comment]
I appreciate that there is no 'world scale', it was the easiest way to communicate the feature. 

Would it be possible to work out what the new local scale would be based on the hierarchy? It's something that we are missing that other engines are able to do :(

**Comment by Maksims** [comment]
Actually, none of the engines can preserve non-uniform scaling when reparenting. This is mathematical problem, not engine one.
There are 2 paths that can "solve" it:
1. Make scale completely independent of hierarchy, so no inheritance of scale, which of course is not good.
2. Restoring scale when reparenting, is possible but only if all ancestors of old and new parent scaled uniformly (same value for each axes, no skewing).

**Comment by karenstritz** [comment]
This is pretty old, but I wanted to chime in as I found the article mentioned above and encountered the same issue. I think @yaustar was referring to simply wanting the ability to reparent uniformly scaled objects and have them stay the same world scale. 

It's expected that non-uniformly scaled objects would cause skewing issues when there are rotations applied, as they do in other engines like Unity. However, for uniformly-scaled objects it is tedious and error-prone having to manually scale things after reparenting. 

If object A has a uniform scale of 2 and is reparented to object B with uniform scale 1.2, the new local scale of reparented object A should change to be originalScale/parentScale or 2/1.2. That would maintain the same world scale for the object without requiring the user to manually calculate and enter three numbers one at a time. 

Note: objects which are reparented to a child of another parent would have to factor in the scales of all previous parents. I'm not sure the exact math, but it seems that multiplying the scales of all previous parents together, having that result be the "parentScale", then dividing originalScale/parentScale again works pretty well. 

You could also store original scale/world scale (scale with no parent) before reparenting, so that you can access it again when reparenting from one child to a different child.
